arkouda/dataframe.py

```diff
@@ -212,6 +213,157 @@ def size(self, as_series=None, sort_index=True):
         else:
             return self._return_agg_dataframe(self.gb.size(), "size", sort_index=sort_index)
 
+    def head(
+        self,
+        n: int = 5,
+        sort_index: bool = True,
+    ) -> DataFrame:
+        """
+        Return the first n rows from each group.
+
+        Parameters
+        ----------
+        n: int, optional, default = 5
+            Maximum number of rows to return for each group.
+            If the number of rows in a group is less than n,
+            all the values from that group will be returned.
+        sort_index: bool, default = True
+            If true, return the DataFrame with indices sorted.
+
+        Returns
+        -------
+        arkouda.dataframe.DataFrame
+
+
+        Examples
+        --------
+        >>> import arkouda as ak
+        >>> from arkouda import *
+        >>> df = ak.DataFrame({"a":ak.arange(10) %3 , "b":ak.arange(10)})
+
+        +----+-----+-----+
+        |    |   a |   b |
+        +====+=====+=====+
+        |  0 |   0 |   0 |
+        +----+-----+-----+
+        |  1 |   1 |   1 |
+        +----+-----+-----+
+        |  2 |   2 |   2 |
+        +----+-----+-----+
+        |  3 |   0 |   3 |
+        +----+-----+-----+
+        |  4 |   1 |   4 |
+        +----+-----+-----+
+        |  5 |   2 |   5 |
+        +----+-----+-----+
+        |  6 |   0 |   6 |
+        +----+-----+-----+
+        |  7 |   1 |   7 |
+        +----+-----+-----+
+        |  8 |   2 |   8 |
+        +----+-----+-----+
+        |  9 |   0 |   9 |
+        +----+-----+-----+
+
+        >>> df.groupby("a").head(2)
+
+        +----+-----+-----+
+        |    |   a |   b |
+        +====+=====+=====+
+        |  0 |   0 |   0 |
+        +----+-----+-----+
+        |  1 |   0 |   3 |
+        +----+-----+-----+
+        |  2 |   1 |   1 |
+        +----+-----+-----+
+        |  3 |   1 |   4 |
+        +----+-----+-----+
+        |  4 |   2 |   2 |
+        +----+-----+-----+
+        |  5 |   2 |   5 |
+        +----+-----+-----+
+
+        """
+        _, indx = self.gb.head(self.df.index.values, n=n, return_indices=True)
+        if sort_index:
+            indx = aksort(indx)
+        return self.df.loc[indx]
```

Comment:
sorry this is on since it's taken me so long to review but I think this needs to be updated now that we've reverted dataframe indexing

Reply:
Fixed now.